boards/atxmega-a1u-xpro/include/eui_provider_params.h

```diff
@@ -0,0 +1,61 @@
+/*
+ * Copyright (C) 2021 Gerson Fernando Budke
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @ingroup     boards_atxmega-a1u-xpro
+ * @{
+ *
+ * @file
+ * @brief       EUI providers found on the board
+ *
+ * @author      Gerson Fernando Budke <nandojve@gmail.com>
+ */
+#ifndef EUI_PROVIDER_PARAMS_H
+#define EUI_PROVIDER_PARAMS_H
+
+#include <string.h>
+
+#include "net/eui64.h"
+#include "periph/cpuid.h"
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/**
+ * @brief    CPUID provides a EUI-64, the same that is printed on the board
+ */
+static inline int _cpuid_get_eui64(uint8_t index, eui64_t *addr)
+{
+    uint8_t cpuid[CPUID_LEN];
+
+    (void)index;
+
+    cpuid_get(cpuid);
+
+    memcpy(addr, cpuid, 8);
+
+    return 0;
+}
```

Comment:
Note that the default `luid` version also generates an EUI-64 from the CPU-ID. It has however the following advantages:

- It will use all bytes of the CPU-ID (via hashing), resulting in a higher entropy. Hence, the chance of two boards having the same EUI is lower. (A CPU ID is only unique when all bytes are used. Usually, boards from the same batch differ only in the last bytes. Buying to boards could easily result in conflicts in the EUI.)
- It will make sure that the EUI generated this was is marked as locally unique. The globally unique EUIs require a world-wide registration to prevent conflicts. I think using EUIs not marked as locally unique only without paying the registration fees can cause legal trouble.
- It will make sure that the EUI is marked as unicast